src/Common/Validation.elm

```diff
@@ -0,0 +1,81 @@
+module Common.Validation exposing
+    ( Codec
+    , CodecError
+    , Positive
+    , compose
+    , getPositive
+    , int
+    , positive
+    , showError
+    , string
+    )
+
+
+type CodecError
+    = CodecError String
+
+
+type alias Codec i o =
+    { decode : i -> Result CodecError o
+    , encode : o -> i
+    , name : String
+    }
```

Comment:
I really like the idea of having codec for the value. I'd probably not tie it to validations though and keep them separated.